src/lib/OS/SyscallHandler.cc

```diff
@@ -1,15 +1,756 @@
 #include "simeng/OS/SyscallHandler.hh"
 
 #include "simeng/OS/SimOS.hh"
+
 namespace simeng {
 namespace OS {
 
-SyscallHandler::SyscallHandler(SimOS* os) : os_(os) {
+SyscallHandler::SyscallHandler(SimOS* OS,
```

Comment:
is memory_ even needed? Would it be better expose a getter function which exposes the reference to the shared ptr? Do you think calling getMemory every time we need access to memory will lead to performance regression?
@FinnWilkinson thoughts?

Reply:
+1

Reply:
Leaving as is